File: services/lucy-dotnet/src/Lucy.AuthPayment.Api/Modules/Payment/Models/PaymentSetting.cs
namespace lucy_auth_payment.Modules.Payment.Models;

public class PaymentSetting
{
    public string PaymentSettingId { get; set; } = null!;
    public string ProviderCode { get; set; } = null!;
    public string ReceiverUserId { get; set; } = null!;
    public string ReceiverName { get; set; } = null!;
    public string ReceiverPhone { get; set; } = null!;
    public string? QrImageUrl { get; set; }
    public string TransferContentTemplate { get; set; } = "LUCY NAP TIEN {ORDER_CODE}";
    public int IsActive { get; set; } = 1;
    public DateTime CreatedAt { get; set; } = DateTime.Now;
    public DateTime? UpdatedAt { get; set; }
}

File: lucy-web-stack/lucy-backend/src/main/java/com/lucy/backend/payment/entity/PaymentSetting.java
package com.lucy.backend.payment.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Entity
@Table(name = "PaymentSettings")
@Data
public class PaymentSetting {
    @Id
    @Column(name = "PaymentSettingId", length = 50)
    private String paymentSettingId;

    @Column(name = "ProviderCode", length = 50, nullable = false)
    private String providerCode;

    @Column(name = "ReceiverUserId", length = 50, nullable = false)
    private String receiverUserId;

    @Column(name = "ReceiverName", length = 100, nullable = false)
    private String receiverName;

    @Column(name = "ReceiverPhone", length = 20, nullable = false)
    private String receiverPhone;

    @Column(name = "QrImageUrl", length = 255)
    private String qrImageUrl;

    @Column(name = "TransferContentTemplate", length = 255, nullable = false)
    private String transferContentTemplate;

    @Column(name = "IsActive", nullable = false)
    private Integer isActive = 1;

    @Column(name = "CreatedAt", nullable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "UpdatedAt")
    private LocalDateTime updatedAt;
}
